src/core/maps/area_occupancy_estimator.h

```diff
@@ -81,9 +81,68 @@ class AreaOccupancyEstimator : public CellOccupancyEstimator {
   virtual Occupancy estimate_occupancy(const Beam &beam,
                                        const Rectangle &cell_bnds,
                                        bool is_occ) override {
-    Intersections intrs = find_intersections(beam, cell_bnds, is_occ);
-    double chunk_area = compute_chunk_area(beam, cell_bnds, is_occ, intrs);
-    return estimate_occupancy(chunk_area, cell_bnds.area(), is_occ);
+    Occupancy result(0.0, 0.0);
+    if (equal (beam.x_st, beam.x_end) && (equal(beam.y_st, beam.y_end))) {
```

Comment:
Space after the first equal is redundant.
Move the condition computation to the separate variable with a meaningful name.